pandas/core/index.py

```diff
@@ -2543,11 +2550,12 @@ def insert(self, loc, item):
         -------
         new_index : Index
         """
-        _self = np.asarray(self)
-        item_idx = Index([item], dtype=self.dtype).values
-        idx = np.concatenate(
-            (_self[:loc], item_idx, _self[loc:]))
-        return Index(idx, name=self.name)
+        indexes=[self[:loc],
+                 Index([item]),
+                 self[loc:]]
+
+        return indexes[0].append(indexes[1]).append(indexes[2])
```

Comment:
not sure I think this is a good idea.

``self._shallow_copy(idx)`` doesn't work?